sentry/src/main/java/io/sentry/protocol/Geo.java

```diff
@@ -0,0 +1,167 @@
+  @Nullable private String countryCode;
+
+  /** Human readable region name or code. * */
+  @Nullable private String region;
+
+  /** unknown fields, only internal usage. */
+  private @Nullable Map<String, @NotNull Object> unknown;
+
+  public Geo() {}
+
+  public Geo(final @NotNull Geo geo) {
+    this.city = geo.city;
+    this.countryCode = geo.countryCode;
+    this.region = geo.region;
+  }
+
+  /**
+   * Gets the human readable city name.
+   *
+   * @return human readable city name
+   */
+  public @Nullable String getCity() {
+    return city;
+  }
+
+  /**
+   * Sets the human readable city name.
+   *
+   * @param city human readable city name
+   */
+  public void setCity(final @Nullable String city) {
+    this.city = city;
+  }
+
+  /**
+   * Gets the two-letter country code (ISO 3166-1 alpha-2).
+   *
+   * @return two-letter country code (ISO 3166-1 alpha-2).
+   */
+  public @Nullable String getCountryCode() {
+    return countryCode;
+  }
+
+  /**
+   * Sets the two-letter country code (ISO 3166-1 alpha-2).
+   *
+   * @param countryCode two-letter country code (ISO 3166-1 alpha-2).
+   */
+  public void setCountryCode(final @Nullable String countryCode) {
+    this.countryCode = countryCode;
+  }
+
+  /**
+   * Gets the human readable region name or code.
+   *
+   * @return human readable region name or code.
+   */
+  public @Nullable String getRegion() {
+    return region;
+  }
+
+  /**
+   * Sets the human readable region name or code.
+   *
+   * @param region human readable region name or code.
+   */
+  public void setRegion(final @Nullable String region) {
+    this.region = region;
+  }
+
+  // region json
+
+  public static final class JsonKeys {
+    public static final String CITY = "city";
+    public static final String COUNTRY_CODE = "country_code";
+    public static final String REGION = "region";
+  }
+
+  @Nullable
+  @Override
+  public Map<String, Object> getUnknown() {
+    return unknown;
+  }
+
+  @Override
+  public void setUnknown(@Nullable Map<String, Object> unknown) {
+    this.unknown = unknown;
+  }
+
+  @Override
+  public void serialize(JsonObjectWriter writer, ILogger logger) throws IOException {
+    writer.beginObject();
+    if (city != null) {
+      writer.name(JsonKeys.CITY).value(city);
+    }
+    if (countryCode != null) {
+      writer.name(JsonKeys.COUNTRY_CODE).value(countryCode);
+    }
+    if (region != null) {
+      writer.name(JsonKeys.REGION).value(region);
+    }
+    if (unknown != null) {
+      for (String key : unknown.keySet()) {
+        Object value = unknown.get(key);
+        writer.name(key);
+        writer.value(logger, value);
+      }
+    }
+    writer.endObject();
+  }
+
+  public static final class Deserializer implements JsonDeserializer<Geo> {
+
+    @Override
+    public Geo deserialize(JsonObjectReader reader, ILogger logger) throws Exception {
+      reader.beginObject();
+      Geo geo = new Geo();
```

Comment:
```suggestion
      final Geo geo = new Geo();
```

We usually like to mark things as much `final` as possible, could you add it here and to other places as well?